src/java/org/apache/cassandra/service/reads/ReadCallback.java

```diff
@@ -56,29 +59,104 @@
 public class ReadCallback<E extends Endpoints<E>, P extends ReplicaPlan.ForRead<E>> implements RequestCallback<ReadResponse>
 {
     protected static final Logger logger = LoggerFactory.getLogger( ReadCallback.class );
-    private class WarningCounter
+
+    private class WarnAbortCounter
     {
-        // the highest number of tombstones reported by a node's warning
-        final AtomicInteger tombstoneWarnings = new AtomicInteger();
-        final AtomicInteger maxTombstoneWarningCount = new AtomicInteger();
-        // the highest number of tombstones reported by a node's rejection. This should be the same as
-        // our configured limit, but including to aid in diagnosing misconfigurations
-        final AtomicInteger tombstoneAborts = new AtomicInteger();
-        final AtomicInteger maxTombstoneAbortsCount = new AtomicInteger();
-
-        // TODO: take message as arg and return boolean for 'had warning' etc
-        void addTombstoneWarning(InetAddressAndPort from, int tombstones)
+        final AtomicInteger warnings = new AtomicInteger();
+        // the highest number reported by a node's warning
+        final AtomicLong maxWarningCount = new AtomicLong();
+
+        final AtomicInteger aborts = new AtomicInteger();
+        // the highest number reported by a node's rejection.
+        final AtomicLong maxAbortsCount = new AtomicLong();
+
+        void addWarning(InetAddressAndPort from, long value)
         {
             if (!waitingFor(from)) return;
-            tombstoneWarnings.incrementAndGet();
-            maxTombstoneWarningCount.accumulateAndGet(tombstones, Math::max);
+            warnings.incrementAndGet();
+            maxWarningCount.accumulateAndGet(value, Math::max);
         }
 
-        void addTombstoneAbort(InetAddressAndPort from, int tombstones)
+        void addAbort(InetAddressAndPort from, long value)
         {
             if (!waitingFor(from)) return;
```

Comment:
See https://github.com/apache/cassandra/pull/1180/files#r701436759

Reply:
your link just loads the diff; not sure what you are linking to

Reply:
removed